oak_restricted_kernel/src/mm/mod.rs

```diff
@@ -307,6 +307,23 @@ pub fn initial_pml4(
         MemoryEncryption::NoEncryption
     };
 
+    // Populate all the kernel space entries of the level 4 page table. This means that these
+    // entries should never change, memory allocation will happen on lower level page tables. This
+    // is done to permit the kernel to switch between different level 4 page tables in the future.
+    // All page tables that include these mappings will point to the same lower level page tables /
+    // memory in kernel space.
+    {
+        let mut fa = FRAME_ALLOCATOR.lock();
+        for entry in pml4.iter_mut().skip(256) {
+            let pml3_frame = fa
```

Comment:
it is unclear to me whether this is a 4KiB or 2MiB phys frame. Could you perhaps annotate the typs of the frame so that we are sure that this doesn use 512MiB?

Reply:
done! they're 4KiB frames